### PR TITLE
Add packet validations support

### PR DESCRIPTION
**Pull request by deitch**
**What this PR does / why we need it**:

Part 2 of 3 in adding packet to the API. First one was #855 , which was just merged in.

Implements [validations](https://github.com/gardener/gardener/blob/master/docs/development/new-cloud-provider-api-extension.md#add-validations) of API extension.

**Release note**:

```improvement operator
Add packet validations. 
```

### CODE REVIEW

File: plugin/pkg/shoot/validator/admission.go [hunk 1]
```diff
@@ -454,6 +467,56 @@ func validateGCP(c *validationContext) field.ErrorList {
 	return allErrs
 }
 
+func validatePacket(c *validationContext) field.ErrorList {
+	var (
+		allErrs = field.ErrorList{}
+		path    = field.NewPath("spec", "cloud", "gcp")
```
Comment (rfranzke) [inline plugin/pkg/shoot/validator/admission.go]:
```suggestion
		path    = field.NewPath("spec", "cloud", "packet")
```

**Review by rfranzke — CHANGES_REQUESTED** [review]
small nit

File: plugin/pkg/shoot/validator/admission.go (hunk 1 above)
Comment (deitch) [reply plugin/pkg/shoot/validator/admission.go]: Oh yes, quite. Fixed.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**